### PR TITLE
164964758 Implement all Model and Migration

### PR DESCRIPTION
**Pull request by ssanusi**
#### Description
The Pull request implement model for user and also 

#### Type of change
added All model to the model directory and 

#### How Has This Been Tested?
run ```npm run db-migrate``` to create user migration 

#### Checklist:
  Implement model and migration for bellow entity 
 - [x] User 
 - [x] Article 
 - [x] Article Rating 
 - [x] Article Report 
 - [x]  Bookmark 
 - [x] Comment
 - [x] Favourite 
 - [x] Follower
 - [x] Highlight
 - [x] Notification

 

#### PT
[#164964758](https://www.pivotaltracker.com/story/show/164964758)

### CODE REVIEW

File: src/migrations/20190401140553-create-article-report.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export default {
+  up: (queryInterface, Sequelize) =>
```
[the remark this answers is not in the document]
Comment (ssanusi) [reply src/migrations/20190401140553-create-article-report.js:2]: fixed

File: src/migrations/20190401140553-create-article-report.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      createdAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
+      },
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
+      },
+    }),
+  down: queryInterface => queryInterface.dropTable('article_reports'),
```
[the remark this answers is not in the document]
Comment (ssanusi) [reply src/migrations/20190401140553-create-article-report.js:26]: fixed

File: src/models/article_report.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { Model } from 'sequelize';
```
[the remark this answers is not in the document]
Comment (ssanusi) [reply src/models/article_report.js:1]: fixed

File: src/migrations/20190329164503-create-user.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      },
+      createdAt: {
+        type: Sequelize.DATE,
+      },
+      updatedAt: {
+        type: Sequelize.DATE,
+      },
+    }),
+
+  down: (queryInterface, Sequelize) => queryInterface.dropTable('users'),
```
[the remark this answers is not in the document]
Comment (ssanusi) [reply src/migrations/20190329164503-create-user.js]: refactored

File: src/models/user.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import { Sequelize, Model } from 'sequelize';
+/**
+ *
+ *
+ * @class User
+ * @extends {Model}
+ */
+class User extends Model {}
+
+export default User.init(
```
[the remark this answers is not in the document]
Comment (ssanusi) [reply src/models/user.js]: fixed

File: src/models/user.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
+import { Sequelize, Model } from 'sequelize';
+/**
+ *
+ *
+ * @class User
+ * @extends {Model}
+ */
+class User extends Model {}
```
[the remark this answers is not in the document]
Comment (ssanusi) [reply src/models/user.js]: refactored

**Review by chikeozulumba — COMMENTED** [review]
LGTM

**Review by toluola — COMMENTED** [review]
I think we should use the squash commit option on Github before merging, that will save us a lot of time

**Review by Simplemart17 — COMMENTED** [review]
From the error log on travis, a dependency package is missing. You might want to fix that for travis to build. Also I want to suggest we squash the commit when merging the PR.

**Pull request merged**